### PR TITLE
@craigspaeth => auction2/auction switcheroo #dontmerge

### PR DESCRIPTION
**Pull request by sweir27**
(I'm just waiting on a final pass of QA on the current version, then hope to get this merged tomorrow, so feel free to wait on looking)

This PR updates the `/auction2` routers to respond to `/auction/` routes, and deletes the `auction` app entirely.

I've gone through all of the basic flows with this implementation (registering, bidding, browsing around, etc.) and have fixed all of the errors I found. I would still like to test this for a little bit in staging once it's merged before deploying.

Also cc @erikdstock I absorbed the change you made in https://github.com/artsy/force/pull/1126 here, as it was causing an especially nefarious bug on the bid form once I switched routing. I agree with @damassi and @craigspaeth that we should come up with a more sustainable solution, but as the issue was blocking this PR I just stuck in your updated regex for now.

### CODE REVIEW

File: desktop/lib/setup.coffee [hunk 1]
```diff
@@ -263,7 +263,6 @@ module.exports = (app) ->
   app.use require "../apps/fair"
   app.use require "../apps/fair_info"
   app.use require "../apps/fair_organizer"
-  app.use require "../apps/auction"
   app.use require "../apps/auction2"
```
Comment (sweir27) [inline desktop/lib/setup.coffee]: A future PR can fix all of the naming (auction2 --> auction), but I thought that would make this diff a little too much to handle.

File: desktop/assets/auctions.coffee [hunk 2]
```diff
@@ -8,8 +8,8 @@ routes =
 
   '''
   /sale/.*
-  /auction/.*
-  ''': require('../apps/auction/client/index.coffee').init
+  ^/auction/[^/]+/?$
```
Comment (sweir27) [inline desktop/assets/auctions.coffee]: cc @erikdstock

File: desktop/apps/auctions/templates/my_active_bids.jade [hunk 3]
```diff
@@ -1,3 +1,3 @@
-if myActiveBids.length
+if myActiveBids && myActiveBids.length
```
Comment (sweir27) [inline desktop/apps/auctions/templates/my_active_bids.jade]: I think this has always thrown an error

File: desktop/assets/auctions.coffee (hunk 2 above)
Comment (damassi) [inline desktop/assets/auctions.coffee]: Would be good if there was a `FIXME:` here linking to the PR in question.

File: desktop/apps/auction2/queries/sale.js [hunk 4]
```diff
@@ -16,10 +16,12 @@ export default function SaleQuery(id) {
       eligible_sale_artworks_count
       end_at
       id
+      is_auction
       is_live_open
       is_open
       live_start_at
       name
+      registration_ends_at
```
Comment (sweir27) [inline desktop/apps/auction2/queries/sale.js]: These two extra fields are necessary for getting the `Registration Closed` state to show up correctly on the registration button.

File: desktop/assets/auctions.coffee [hunk 5]
```diff
@@ -6,10 +6,12 @@ routes =
...
   ''': require('../apps/auctions/client/index.coffee').init
 
+  # TODO: refactor to check paths in order.
+  # Discussion here: https://github.com/artsy/force/pull/1126
   '''
   /sale/.*
-  /auction/.*
-  ''': require('../apps/auction/client/index.coffee').init
+  ^/auction/[^/]+/?$
+  ''': -> require('../apps/auction2/client/index.js').default
```
Comment (craigspaeth) [inline desktop/assets/auctions.coffee]: I think you don't want the arrow here—that's wrapping the code in an anonymous function and not actually calling the `default` function.

**Comment by craigspaeth** [comment]
w00t w00t! Congrats on this, excited to have this merged! The only blocker is my small note about the wrapping arrow, and you already mentioned following up with the rest of the renaming. Well done @sweir27 and @damassi 👏  👏  👏

File: desktop/assets/auctions.coffee (hunk 5 above)
Comment (sweir27) [inline desktop/assets/auctions.coffee]: 😱 didn't even see that! Thanks!

File: desktop/apps/auction2/client/index.js [hunk 6]
```diff
@@ -11,60 +11,62 @@ import { setupCommercialFilter } from './commercial_filter'
 import { data as sd } from 'sharify'
 import _ from 'underscore'
 
-const auction = new Auction(sd.AUCTION)
-const user = sd.CURRENT_USER ? new CurrentUser(sd.CURRENT_USER) : null
+export default () => {
```
Comment (sweir27) [inline desktop/apps/auction2/client/index.js]: This diff is a little weird, but all I did was explicitly export the code here so we could require it in the assets file.

File: desktop/apps/auction2/client/index.js [hunk 7]
```diff
@@ -11,60 +11,62 @@ import { setupCommercialFilter } from './commercial_filter'
...
-  setupCommercialFilter()
-}
+  // render the associated auction if there is one
+  if (auction.get('associated_sale')) {
+    const associatedSale = auction.get('associated_sale')
+    mountAuctionBlock({
+      relatedAuction: true,
+      sale: associatedSale
+    }, '#associated-sale')
+  }
```
Comment (damassi) [inline desktop/apps/auction2/client/index.js]: Super minor, but why not 

```javascript
const sale = auction.get('associated_sale')

if (sale) {
  mountAuctionBlock({
    relatedAuction: true,
    sale 
  })
}
```

(Having `relatedAuction` and `associatedSale` seems a bit confusing, like they're referring to the same different thing.)

File: desktop/assets/auctions.coffee (hunk 5 above)
Comment (mzikherman) [inline desktop/assets/auctions.coffee]: literally just fixed that exact thing for professional buyer's asset package! https://github.com/artsy/force/pull/1172/files#diff-7c9d6756206ba48b2a2cb3293ea995e5L19

**Comment by sweir27** [comment]
Gonna merge this to test it out in staging!

**Pull request merged**